File: LeetCode-Java/Binary Tree Inorder Traversal .java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
import java.util.Stack;
import java.util.ArrayList;
public class Solution 
{
    public List<Integer> inorderTraversal(TreeNode root) 
    {
        List<Integer> resultList = new ArrayList<Integer>();
        if(root == null) return resultList;
        Stack<TreeNode> treeStack = new Stack<TreeNode>();
        treeStack.push(root);
        root = root.left;
        while(root != null || !treeStack.empty())
        {
            if(root != null)
            {
                treeStack.push(root);
                root = root.left;
            }
            else
            {
                TreeNode tempNode = treeStack.pop();
                resultList.add(tempNode.val);
                root = tempNode.right;
            }
        }
        return resultList;
    }
}

File: Leetcode-cpp/binary_tree_inorder_traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
 public:
  vector<int> inorderTraversal(TreeNode* root) {
    vector<int> result;
    stack<TreeNode*> nodeStack;
    while (root != nullptr) {
      nodeStack.push(root);
      root = root->left;
    }
    while (!nodeStack.empty()) {
      TreeNode* tempNode = nodeStack.top();
      nodeStack.pop();
      result.push_back(tempNode->val);
      tempNode = tempNode->right;
      while (tempNode != nullptr) {
        nodeStack.push(tempNode);
        tempNode = tempNode->left;
      }
    }
    return result;
  }
};